entities.py

```diff
@@ -155,11 +155,14 @@ def use_card(self, card, exhaust, pile, enemies, target: "Enemy"=None) -> None:
         sleep(0.5)
         view.clear()
 
-    def draw_cards(self, middle_of_turn: bool=True, cards: int = 0):
+    def draw_cards(self, middle_of_turn: bool=True, cards: int = -1):
         """Draws [draw_cards] cards."""
         _ = middle_of_turn
-        if cards == 0:
+        if cards == -1:
             cards = self.draw_strength
+        if helper.NoDraw in self.debuffs:
+            ansiprint("You cannot draw any cards because of <debuff>No Draw</debuff>.")
+            return
```

Comment:
No need to split up the NoDraw code, especially because there might be other effects that affect the draw strength.

Instead, publish a `BEFORE_DRAW` message with all the info needed, like the player and the amount that will be drawn, and in the NoDraw effect, catch the message and modify or cancel the pending action. 

That way, NoDraw and other draw effects can keep their code together, and the draw function just says "Hey, I'm about to do this, does anybody care?" -- and other things can modify it as needed. You could even have something simple like:

```python
class PendingAction():
```

with methods to cancel or modify the action. So draw_cards becomes "Create pending action. Publish it. (Action gets modified/cancelled). Execute it."

Reply:
How would the `PendingAction` class be implemented? Does it take a function and then control if it runs from inside the class?